diff --git a/PKGBUILD b/PKGBUILD
index d811e23..df50b83 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -28,8 +28,6 @@ makedepends=(
   graphviz
   texlive-bin
   texlive-latexextra
-  # -cuda
-  cuda
   # -hip
   hip-runtime-amd
   hipblas
@@ -64,7 +62,7 @@ build() {
     -DGINKGO_BUILD_BENCHMARKS=ON
     -DGINKGO_BUILD_EXAMPLES=ON
     -DGINKGO_BUILD_DOC=ON
-    -DGINKGO_BUILD_TESTS=ON
+    -DGINKGO_BUILD_TESTS=OFF # Workaround for CMake error with misssing test name
   )
   # In general, we want to list all real archs (sm_XX) and the latest virtual arch (compute_XX) for future PTX compatibility.
   # Valid values can be discovered from nvcc --help
@@ -79,6 +77,7 @@ build() {
     -DGINKGO_BUILD_SYCL=OFF
   cmake --build build
 
+  : <<COMMENT_SEPARATOR
   # -cuda package
   # Compile source code for supported GPU archs in parallel
   export CUDAFLAGS="--threads $(nproc)"
@@ -88,6 +87,7 @@ build() {
     -DGINKGO_BUILD_HIP=OFF \
     -DGINKGO_BUILD_SYCL=OFF
   cmake --build build-cuda
+COMMENT_SEPARATOR
 
   # -hip package
   # LTO does not work with HIP
@@ -100,7 +100,6 @@ build() {
   # Compile source code for supported GPU archs in parallel
   export HIPFLAGS="$HIPFLAGS -parallel-jobs=$(nproc)"
   cmake -B build-hip "${common_cmake_flags[@]}" \
-    -DCMAKE_CXX_COMPILER=/opt/rocm/lib/llvm/bin/amdclang++ \
     -DCMAKE_HIP_ARCHITECTURES="$_amdgpu_archs" \
     -DGINKGO_BUILD_CUDA=OFF \
     -DGINKGO_BUILD_HIP=ON \
@@ -186,4 +185,6 @@ package_ginkgo-hpc-hip() {
   _package -hip
 }
 
+pkgname=(${pkgname[@]/ginkgo-hpc-cuda})
+
 # vim:set ts=2 sw=2 et:
