diff --git a/PKGBUILD b/PKGBUILD
index 8216f67..16378ff 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -140,6 +140,9 @@ prepare() {
   git -C release/datafiles/assets lfs checkout
 
   patch -p1 -i "$srcdir"/hiprt-lib-path.patch
+
+  # Support loong64
+  patch -p1 -i "$srcdir"/add-loong64-support.patch
 }
 
 _get_pyver() {
@@ -160,7 +163,6 @@ build() {
     -D CMAKE_BUILD_TYPE=Release
     -D CMAKE_INSTALL_PREFIX=/usr
     -D WITH_LINKER_MOLD=ON
-    -D CUDA_HOST_COMPILER="$NVCC_CCBIN"
     -D HIP_ROOT_DIR=/opt/rocm
     -D HIPRT_INCLUDE_DIR=/opt/rocm/include
     -D OCLOC_INSTALL_DIR=/usr
@@ -173,6 +175,13 @@ build() {
     -D WITH_CYCLES_OSL=ON
     -D WITH_INSTALL_PORTABLE=OFF
     -D WITH_PYTHON_INSTALL=OFF
+    -D WITH_CYCLES_EMBREE=OFF
+    -D WITH_CYCLES_DEVICE_CUDA=OFF
+    -D WITH_CYCLES_DEVICE_OPTIX=OFF
+    -D WITH_CYCLES_CUDA_BINARIES=OFF
+    -D WITH_CUDA_DYNLOAD=OFF
+    -D WITH_CYCLES_DEVICE_HIP=OFF
+    -D WITH_CYCLES_HIP_BINARIES=OFF
     -G Ninja
     -S "$pkgname"
     -W no-dev
@@ -194,7 +203,13 @@ package() {
 
   # Move OneAPI AOT lib to proper place
   mkdir "${pkgdir}"/usr/lib/
-  mv "${pkgdir}"/usr/share/blender/lib/libcycles_kernel_oneapi_aot.so "${pkgdir}"/usr/lib/
+  # mv "${pkgdir}"/usr/share/blender/lib/libcycles_kernel_oneapi_aot.so "${pkgdir}"/usr/lib/
 
   install -vDm 644 doc/license/{BSD-{2,3}-Clause,MIT,Zlib}-license.txt -t "$pkgdir/usr/share/licenses/$pkgname/"
 }
+
+source+=("add-loong64-support.patch::https://projects.blender.org/blender/blender/commit/d18a0cf277e3d7403eb18718b9cb7a6ec036819f.patch")
+sha512sums+=('4bcbe578638dec0d7d3808835d7191f890c7060adbd6b4006e1e1fc4eee70ae403527419c6583913c43254d5ca8dbf6427ba9b445d45ac5f4af7e5a26b88c98b')
+depends=($(printf "%s\n" "${depends[@]}" | grep -Ev '^(embree|intel-oneapi-compiler-dpcpp-cpp-runtime-libs|intel-oneapi-compiler-shared-runtime-libs|openimagedenoise|usd)$'))
+makedepends=($(printf "%s\n" "${makedepends[@]}" | grep -Ev '^(cuda|intel-oneapi-compiler-shared-runtime|intel-oneapi-dpcpp-cpp|intel-compute-runtime)$'))
+makedepends+=(vulkan-headers)
